src/python/lib/ensembl/compara/citest/testdb.py

```diff
@@ -0,0 +1,215 @@
+
+        If `group_by` is provided, the number of rows will be compared per group, applying the same variation
+        to all of them. If `filter_by` is provided, only the rows matching all the given conditions will be
+        compared.
+
+        Args:
+            variation: Allowed variation between reference and target tables.
+            group_by: Group rows by column(s), and count the number of rows per group.
+            filter_by: Filter rows by one or more conditions (joined by the AND operator).
+
+        Raise:
+            FailedDBTestException: If `group_by` is provided and the number of groups is different; or if the
+                number of rows differ for at least one group.
+
+        """
+        # Compose the sql query from the given parameters
+        sql_filter = self._get_sql_filter(filter_by)
+        if group_by:
+            if isinstance(group_by, str):
+                group_by = [group_by]
+            # ORDER BY to ensure that the results are always in the same order (for the same groups)
+            sql_query = "SELECT `{0}`, COUNT(*) as nrows FROM {1} {2} GROUP BY `{0}` ORDER BY `{0}`".format(
+                "`,`".join(group_by), self.table, sql_filter)
+        else:
+            sql_query = "SELECT COUNT(*) as nrows FROM {} {}".format(self.table, sql_filter)
+        # Get the number of rows for both databases
+        ref_data = pandas.read_sql_query(sql_query, self.ref_db.bind)
+        target_data = pandas.read_sql_query(sql_query, self.target_db.bind)
+        # Check if the size of the returned tables are the same
+        if ref_data.shape != target_data.shape:
+            expected = ref_data.shape[0]
+            found = target_data.shape[0]
+            # Note: the shape can only be different if group_by is given
+            message = "Different number of groups ({}) for table '{}'".format(", ".join(group_by), self.table)
+            raise FailedDBTestException(expected, found, sql_query, message)
+        # Check if the number of rows (per group) are the same
+        difference = abs(ref_data['nrows'] - target_data['nrows'])
+        allowed_variation = ref_data['nrows'] * variation
+        failing_rows = difference > allowed_variation
+        if failing_rows.any():
+            expected_data = ref_data.loc[failing_rows]
+            expected = [] if expected_data.empty else expected_data.to_string(index=False).splitlines()
+            found_data = target_data.loc[failing_rows]
+            found = [] if found_data.empty else found_data.to_string(index=False).splitlines()
+            message = ("The difference in number of rows for table '{}' exceeds the allowed variation "
+                       "({})").format(self.table, variation)
+            raise FailedDBTestException(expected, found, sql_query, message)
+
+    def test_content(self, *, columns: Union[str, List] = None, ignore_columns: Union[str, List] = None,
```

Comment:
What does the star mean ? Are you using it to discard some arguments ?

Reply:
Putting the `*` forces to always call this method by using keyword arguments (e.g. `mytest.test_content(columns=['a','b'], filter_by=['a > 5'])`), as any no-named argument is consumed by `*`. This is mainly to force a better readability of the code, which is advisory when having mutually exclusive parameters ([link](https://www.vegardstikbakke.com/python-keyword-only/)).